frontend/Interfaces/buddy/DIP/ImageContainer.h

```diff
@@ -22,24 +22,254 @@
 #define FRONTEND_INTERFACES_BUDDY_DIP_IMAGECONTAINER
 
 #include "buddy/Core/Container.h"
+#include "buddy/DIP/imgcodecs/Replenishment.hpp"
 #include <cassert>
 #include <opencv2/opencv.hpp>
 
+using namespace dip;
 // Image container.
 // - T represents the type of the elements.
 // - N represents the number of dimensions.
-// - image represents the OpenCV Mat object.
-// - norm indicates whether to perform normalization, and the normalization is
-//   disabled by default.
 template <typename T, size_t N> class Img : public MemRef<T, N> {
 public:
+  Img();
+
+  /**
+   * @brief overload
+   * @param rows Number of rows in a 2D array.
+   * @param cols Number of columns in a 2D array.
+   * @param type Array type. Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  Img(int rows, int cols, int type);
+
+  /**
+   * @brief overload
+   * @param rows Number of rows in a 2D array.
+   * @param cols Number of columns in a 2D array.
+   * @param type Array type. Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   * @param data Pointer to the user data.
+   * they just initialize the matrix header that points to the specified data.
+   */
+  Img(int rows, int cols, int type, T *data);
+
+  /**
+   * @brief overload
+   * @param ndims Array dimensionality.
+   * @param sizes Array of integers specifying an n-dimensional array shape.
+   * @param type Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  Img(int ndims, intptr_t *sizes, int type);
+
+  /**
+   * @brief overload
+   * @param m Array that (as a whole or partly) is assigned to the constructed
+   * matrix.
+   */
+  Img(const Img<T, N> &m);
+
+  /**
+   * @brief assignment operators
+   * @param m Assigned, right-hand-side matrix.
+   * matrix.
+   */
+  Img &operator=(const Img<T, N> &m);
+
+  /**
+   * @brief Load image data from OpenCV Mat.
+   * @param image represents the OpenCV Mat object.
+   * @param norm indicates whether to perform.
+   */
   Img(cv::Mat image, intptr_t sizes[N] = nullptr, bool norm = false);
 
```

Comment:
Please add copy constructor and move constructor.

Reply:
> Please add move constructor.

ok.